pandas/tests/test_timedelta64_overflow.py

```diff
@@ -0,0 +1,322 @@
+    add_sub_types: BinaryOpTypes,
+    positive_td: Timedelta,
+):
+    left = wrap_value(Timedelta.max, add_sub_types.left)
+    right = wrap_value(positive_td, add_sub_types.right)
+
+    if add_sub_types.result is Timedelta:
+        msg = "|".join(
+            [
+                "int too big to convert",
+                "Python int too large to convert to C long",
+            ]
+        )
+    else:
+        msg = "Overflow in int64 addition"
+
+    with pytest.raises(OverflowError, match=msg):
+        left + right
+
+    with pytest.raises(OverflowError, match=msg):
+        right + left
+
+
+@xfail_no_overflow_check
+@given(positive_td=positive_tds)
+def test_sub_raises_expected_error_if_result_would_overflow(
+    add_sub_types: BinaryOpTypes,
+    positive_td: Timedelta,
+):
+    left = wrap_value(Timedelta.min, add_sub_types.left)
+    right = wrap_value(positive_td, add_sub_types.right)
+
+    msg = "Overflow in int64 addition"
+    with pytest.raises(OverflowError, match=msg):
+        left - right
+
+    with pytest.raises(OverflowError, match=msg):
+        (-1 * right) - abs(left)
+
+
+@given(td_value=positive_tds)
+def test_add_timestamp_raises_expected_error_if_result_would_overflow(
+    ts_add_sub_types: BinaryOpTypes,
+    td_value: Timedelta,
+):
+    left = wrap_value(td_value, ts_add_sub_types.left)
+    right = wrap_value(Timestamp.max, ts_add_sub_types.right)
+
+    ex = (OutOfBoundsDatetime, OverflowError)
+    msg = "|".join(["Out of bounds nanosecond timestamp", "Overflow in int64 addition"])
+
+    with pytest.raises(ex, match=msg):
+        left + right
+
+    with pytest.raises(ex, match=msg):
+        right + left
+
+
+@xfail_no_overflow_check
+@given(td_value=positive_tds)
+def test_sub_timestamp_raises_expected_error_if_result_would_overflow(
+    ts_add_sub_types: BinaryOpTypes,
+    td_value: Timedelta,
+):
+    right = wrap_value(td_value, ts_add_sub_types[0])
+    left = wrap_value(Timestamp.min, ts_add_sub_types[1])
+
+    ex = (OutOfBoundsDatetime, OverflowError)
+    msg = "|".join(["Out of bounds nanosecond timestamp", "Overflow in int64 addition"])
+
+    with pytest.raises(ex, match=msg):
+        left - right
+
+
+@given(value=st.floats().filter(lambda f: abs(f) > 1))
+def test_scalar_multiplication_raises_expected_error_if_result_would_overflow(
+    value: float,
+):
+    td = Timedelta.max
+
+    msg = "|".join(
+        [
+            "cannot convert float infinity to integer",
+            "Python int too large to convert to C long",
+            "int too big to convert",
+        ]
+    )
+    with pytest.raises(OverflowError, match=msg):
+        td * value
+
+    with pytest.raises(OverflowError, match=msg):
+        value * td
+
+
+@xfail_no_overflow_check
+@given(value=st.floats().filter(lambda f: abs(f) > 1))
+@pytest.mark.parametrize(
+    argnames="td_type",
+    argvalues=timedelta_types[containers],  # type: ignore[arg-type]
```

Comment:
this is a pattern im not familar with -> mental overhead.  is this useful enough to ask reviewers/future-contributors to learn it?  or can we use a more plain-vanilla pattern?

Reply:
Oh no, I'm not trying to advocate for a new pattern (think this just matched my mental model at the time). I'll rewrite this to better follow existing conventions.